src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -310,6 +310,67 @@ def table_function_partition_spec(
     )
 
 
+def indent_child_query(child: str) -> str:
```

Comment:
To clarify, this is just a formatter that beautifies the SQL with changing any semantic right and it's mostly helpful for local debugging purpose?

have we thought about use existing libraries to do formatting?
https://github.com/tobymao/sqlglot
https://github.com/andialbrecht/sqlparse

(add dependency or vendor their source code depending on their license)